podpac/core/cache/cache.py

```diff
@@ -753,6 +761,12 @@ def put(self, node, data, key, coordinates=None, update=False):
         # if file for listing does not already exist, we need to create a new container, add the listing, and save to file
         else:
             self.save_new_container(listings=[listing], path=path)
+
+        if self.max_size is not None and self.size() >= self.max_size:
+        #     # TODO removal policy
+            self.rem(node=node, key=key, coordinates=coordinates)
+            raise CacheException("Cache is full. Remove some old entries and try again.")
```

Comment:
This exception (or warning) should advise the user to increase the cache size. Maybe it could suggest both increasing the cache size or removing some old entries. A "full" RAM cache could have no entries in the cache to remove!

Reply:
Thanks @jmilloy . These are good points. I agree the cache should issue a warning and not an error, and I think the suggestion should be to increase the cache size. 

Specifically, I think that the suggestion should be to modify the limit in the settings module and that the cache store classes should be changed so that the limit is dynamically looked up from the settings module - unless an over-ride limit was given when the cache store object was created (e.g. to `__init__()`). Does this make sense and sound like a good plan?

Reply:
Maybe the warning should also document how we ended up implementing the process RAM limitation. Something like
`"Warning: Process is using more RAM than the specified limit in settings.MAX_RAM_CACHE. No longer caching. Consider increasing this limit and restarting the Python process or try clearing the cache (e.g. node.rem_cache(key='*', mode='RAM', all_cache=True) to clear ALL cached results in RAM)"`

If you can easily do a dynamic lookup of the settings that would be preferred over having to restart the process. But otherwise, just make the warning informative.